node/NetworkConfig.cpp

```diff
@@ -424,6 +425,11 @@ bool NetworkConfig::fromDictionary(const Dictionary<ZT_NETWORKCONFIG_DICT_CAPACI
 					if (d.get(ZT_NETWORKCONFIG_DICT_KEY_CLIENT_ID, this->ssoClientID, (unsigned int)sizeof(this->ssoClientID)) > 0) {
 						this->ssoClientID[sizeof(this->ssoClientID) - 1] = 0;
 					}
+					if (d.get(ZT_NETWORKCONFIG_DICT_KEY_SSO_PROVIDER, this->ssoProvider, (unsigned int)(sizeof(this->ssoProvider))) > 0) {
+						this->ssoProvider[sizeof(this->ssoProvider) - 1] = 0;
+					} else {
+						strncpy(this->ssoProvider, "default", sizeof(this->ssoProvider));
```

Comment:
So `strncpy` doesn't actually guarantee that the char[] ends in a null since it will fill the entire array with characters and then stop. Yet another way that C string handling is about as safe as untethered free climbing while drunk. Need to `this->ssoProvider[sizeof(this->ssoProvider) - 1] = 0` or something like that.

Reply:
Actually this particular line might be fine because "default" always fits.

Reply:
Went ahead & added the null anyway